### PR TITLE
Add sized-based split weights for Hudi connector

### PR DESCRIPTION
**Pull request by yihua**
### Change Logs

This PR adds split weight to HudiSplit based on the split size in Hudi connector.  Before this change, the Hudi split weight is uniform, set to default 1.0, which limits the number of splits scheduled to be processed (queued + running) to 100. This becomes the bottleneck for the query performance when the splits are small (e.g. a few MB). The dynamic weight assignment allows much more splits to be scheduled to run, better using the compute and memory resources.

When using size-based split weight, the weight value is between a minimum configured weight and 1.  Given the split size in bytes as `splitSizeInBytes`, the raw weight is calculated as `splitSizeInBytes / standardSplitSize` and then capped in the above range.  There are three new properties introduced (defaults should work well):
- `hudi.size-based-split-weights-enabled` (default: `true`): whether to enable size-based split weight in Hudi connector
- `hudi.standard-split-weight-size` (default: `128MB`): the split size corresponding to the standard split weight when size-based split weight is enabled
- `hudi.minimum-assigned-split-weight` (default: `0.05`): the minimum split weight when size-based split weight is enabled

The concept of using size-based split weight for Hudi split in Presto is similar to Trino Hudi connector, implemented [here](https://github.com/codope/trino/pull/15).

### Test plan

The changes are benchmarked through TPC-DS queries using Hudi connector.  The Presto cluster consists of 1 coordinator and 10 workers in `r5.4xlarge` type.  This PR improves the query latency of selective TPC-DS 1TB queries by ~10%.

```
== RELEASE NOTES ==

General Changes
- Add sized-based split weights for Hudi connector to improve query performance
```

### CODE REVIEW

File: presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java [hunk 1]
```diff
@@ -144,7 +154,9 @@ private ExtendedFileSystem getFileSystem(ConnectorSession session, HudiTableHand
         }
     }
 
-    private Optional<HudiSplit> createHudiSplit(HudiTableHandle table, FileSlice slice, String timestamp, HudiPartition partition)
+    private Optional<HudiSplit> createHudiSplit(
+            HudiTableHandle table, FileSlice slice, String timestamp, HudiPartition partition,
```
Comment (7c00) [inline presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java:158]: nit: for long argument list, we tend to put one argument per line.

File: presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java [hunk 2]
```diff
@@ -120,7 +129,8 @@ public ConnectorSplitSource getSplits(
             Path partitionPath = new Path(hudiPartition.getStorage().getLocation());
             String relativePartitionPath = FSUtils.getRelativePartitionPath(tablePath, partitionPath);
             fsView.getLatestFileSlicesBeforeOrOn(relativePartitionPath, timestamp, false)
-                    .map(fileSlice -> createHudiSplit(table, fileSlice, timestamp, hudiPartition))
+                    .map(fileSlice -> createHudiSplit(
+                            table, fileSlice, timestamp, hudiPartition, splitWeightProvider))
```
Comment (7c00) [inline presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java:133]: nit: one argument per line

**Review by 7c00 — APPROVED** [review]
LGTM.

File: presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSessionProperties.java [hunk 3]
```diff
@@ -38,6 +43,10 @@
     private static final String PARQUET_BATCH_READ_OPTIMIZATION_ENABLED = "parquet_batch_read_optimization_enabled";
     private static final String PARQUET_BATCH_READER_VERIFICATION_ENABLED = "parquet_batch_reader_verification_enabled";
 
+    public static final String SIZE_BASED_SPLIT_WEIGHTS_ENABLED = "size_based_split_weights_enabled";
+    private static final String STANDARD_SPLIT_WEIGHT_SIZE = "standard_split_weight_size";
```
[the remark this answers is not in the document]
Comment (yihua) [reply presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSessionProperties.java:47]: Addressed.  Made them all private.

File: presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java (hunk 1 above)
Comment (yihua) [reply presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java:158]: Fixed.

File: presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java (hunk 2 above)
Comment (yihua) [reply presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java:133]: Given that the line is not long enough, I put them into one line with the lambda function.

**Pull request merged**